fastapi/routing.py

```diff
@@ -1328,30 +1356,37 @@ def read_users():
                     generate_unique_id_function=current_generate_unique_id,
                 )
             elif isinstance(route, routing.Route):
+                path = route.path
+                if self.ignore_trailing_slash:
+                    path = path.rstrip("/")
```

Comment:
Just as a quick drive-by comment, I think it would be good to find a way to avoid this repetition of the `if` check and the `rstrip` functionality. Perhaps we can move it into a small helper function `_normalize_path` or something such so that we can do 
```
path = self._normalize_path(router)
```
and avoid the code duplication.

Reply:
Thank you for the feedback. I introduced two helper methods `_normalize_path` and `_normalize_route` to reduce code duplication.